### ISSUE TITLE
Fatal error when calling MPI_Win_allocate with MPI_Info handle

### ISSUE THREAD

**Issue by sekoenig** [open-mpi/ompi#9721]
## Background information

In my code I am using an `MPI_Win` with `MPI_Accumulate`.  I would like to tweak the algorithm by specifying hints via an `MPI_Info` handle. When I do this, my code aborts with a fatal error.  The same code passing `MPI_INFO_NULL` to `MPI_Win_allocate` runs without error. I include a test program below.

### What version of Open MPI are you using? (e.g., v3.0.5, v4.0.2, git branch name and hash, etc.)

OpenMPI 4.1.1.

### Describe how Open MPI was installed (e.g., from a source/distribution tarball, from a git clone, from an operating system distribution package, etc.)

Deployed by administrators of HPC system. See below for complete system information.

### Please describe the system on which you are running

Output from `ompi_info`:
```
                 Package: Open MPI swmanage@jwlogin07.juwels Distribution
                Open MPI: 4.1.1
  Open MPI repo revision: v4.1.1
   Open MPI release date: Apr 24, 2021
                    OPAL: 4.1.1
      OPAL repo revision: v4.1.1
       OPAL release date: Apr 24, 2021
                 MPI API: 3.1.0
            Ident string: 4.1.1
                  Prefix: /p/software/juwels/stages/2020/software/OpenMPI/4.1.1-intel-compilers-2021.2.0-GCC-10.3.0
 Configured architecture: x86_64-pc-linux-gnu
          Configure host: jwlogin07.juwels
           Configured by: swmanage
           Configured on: Wed Jun  9 13:33:11 UTC 2021
          Configure host: jwlogin07.juwels
  Configure command line: '--prefix=/p/software/juwels/stages/2020/software/OpenMPI/4.1.1-intel-compilers-2021.2.0-GCC-10.3.0' '--build=x86_64-pc-linux-gnu' '--host=x86_64-pc-linux-gnu' '--enable-shared' '--with-hwloc=/p/software/juwels/stages/2020/software/hwloc/2.4.1-GCCcore-10.3.0' '--with-ucx=/p/software/juwels/stages/2020/software/UCX/1.10.1' '--with-verbs' '--with-libevent=/p/software/juwels/stages/2020/software/libevent/2.1.12-GCCcore-10.3.0' '--without-orte' '--without-psm2' '--disable-oshmem' '--with-cuda=/p/software/juwels/stages/2020/software/CUDA/11.3' '--with-ime=/opt/ddn/ime' '--with-gpfs' '--with-slurm' '--with-pmix=external' '--with-libevent=external' '--with-ompi-pmix-rte'
                Built by: swmanage
                Built on: Wed Jun  9 13:53:29 UTC 2021
              Built host: jwlogin07.juwels
              C bindings: yes
            C++ bindings: no
             Fort mpif.h: yes (all)
            Fort use mpi: yes (full: ignore TKR)
       Fort use mpi size: deprecated-ompi-info-value
        Fort use mpi_f08: yes
 Fort mpi_f08 compliance: The mpi_f08 module is available, but due to limitations in the ifort compiler and/or Open MPI, does not support the following: array subsections, direct passthru (where possible) to underlying Open MPI's C functionality
  Fort mpi_f08 subarrays: no
           Java bindings: no
  Wrapper compiler rpath: runpath
              C compiler: icc
     C compiler absolute: /p/software/juwels/stages/2020/software/intel-compilers/2021.2.0-GCC-10.3.0/compiler/2021.2.0/linux/bin/intel64/icc
  C compiler family name: INTEL
      C compiler version: 2021.20210228
            C++ compiler: icpc
   C++ compiler absolute: /p/software/juwels/stages/2020/software/intel-compilers/2021.2.0-GCC-10.3.0/compiler/2021.2.0/linux/bin/intel64/icpc
           Fort compiler: ifort
       Fort compiler abs: /p/software/juwels/stages/2020/software/intel-compilers/2021.2.0-GCC-10.3.0/compiler/2021.2.0/linux/bin/intel64/ifort
         Fort ignore TKR: yes (!DEC$ ATTRIBUTES NO_ARG_CHECK ::)
   Fort 08 assumed shape: yes
      Fort optional args: yes
          Fort INTERFACE: yes
    Fort ISO_FORTRAN_ENV: yes
       Fort STORAGE_SIZE: yes
      Fort BIND(C) (all): yes
      Fort ISO_C_BINDING: yes
 Fort SUBROUTINE BIND(C): yes
       Fort TYPE,BIND(C): yes
 Fort T,BIND(C,name="a"): yes
            Fort PRIVATE: yes
          Fort PROTECTED: yes
           Fort ABSTRACT: yes
       Fort ASYNCHRONOUS: yesMPI_Win_allocate
          Fort PROCEDURE: yes
         Fort USE...ONLY: yes
           Fort C_FUNLOC: yes
 Fort f08 using wrappers: yes
         Fort MPI_SIZEOF: yes
             C profiling: yes
           C++ profiling: no
   Fort mpif.h profiling: yes
  Fort use mpi profiling: yes
   Fort use mpi_f08 prof: yes
          C++ exceptions: no
          Thread support: posix (MPI_THREAD_MULTIPLE: yes, OPAL support: yes, OMPI progress: no, Event lib: yes)
           Sparse Groups: no
  Internal debug support: no
  MPI interface warnings: yes
     MPI parameter check: runtime
Memory profiling support: no
Memory debugging support: no
              dl support: yes
   Heterogeneous support: no
       MPI_WTIME support: native
     Symbol vis. support: yes
   Host topology support: yes
            IPv6 support: no
      MPI1 compatibility: no
          MPI extensions: affinity, cuda, pcollreq
   FT Checkpoint support: no (checkpoint thread: no)
   C/R Enabled Debugging: no
  MPI_MAX_PROCESSOR_NAME: 256
    MPI_MAX_ERROR_STRING: 256
     MPI_MAX_OBJECT_NAME: 64
        MPI_MAX_INFO_KEY: 36
        MPI_MAX_INFO_VAL: 256
       MPI_MAX_PORT_NAME: 1024
  MPI_MAX_DATAREP_STRING: 128
           MCA allocator: basic (MCA v2.1.0, API v2.0.0, Component v4.1.1)
           MCA allocator: bucket (MCA v2.1.0, API v2.0.0, Component v4.1.1)
           MCA backtrace: execinfo (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA btl: vader (MCA v2.1.0, API v3.1.0, Component v4.1.1)
                 MCA btl: tcp (MCA v2.1.0, API v3.1.0, Component v4.1.1)
                 MCA btl: self (MCA v2.1.0, API v3.1.0, Component v4.1.1)
                 MCA btl: openib (MCA v2.1.0, API v3.1.0, Component v4.1.1)
                 MCA btl: smcuda (MCA v2.1.0, API v3.1.0, Component v4.1.1)
            MCA compress: gzip (MCA v2.1.0, API v2.0.0, Component v4.1.1)
            MCA compress: bzip (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA crs: none (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA dl: dlopen (MCA v2.1.0, API v1.0.0, Component v4.1.1)
               MCA event: external (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA hwloc: external (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA if: linux_ipv6 (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA if: posix_ipv4 (MCA v2.1.0, API v2.0.0, Component v4.1.1)
         MCA installdirs: env (MCA v2.1.0, API v2.0.0, Component v4.1.1)
         MCA installdirs: config (MCA v2.1.0, API v2.0.0, Component v4.1.1)
              MCA memory: patcher (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA mpool: hugepage (MCA v2.1.0, API v3.0.0, Component v4.1.1)
             MCA patcher: overwrite (MCA v2.1.0, API v1.0.0, Component v4.1.1)
                MCA pmix: ext3x (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA pmix: flux (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA pmix: isolated (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA pstat: linux (MCA v2.1.0, API v2.0.0, Component v4.1.1)
              MCA rcache: rgpusm (MCA v2.1.0, API v3.3.0, Component v4.1.1)
              MCA rcache: gpusm (MCA v2.1.0, API v3.3.0, Component v4.1.1)
              MCA rcache: grdma (MCA v2.1.0, API v3.3.0, Component v4.1.1)
           MCA reachable: netlink (MCA v2.1.0, API v2.0.0, Component v4.1.1)
           MCA reachable: weighted (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA shmem: posix (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA shmem: sysv (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA shmem: mmap (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA timer: linux (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA bml: r2 (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: self (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: libnbc (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: inter (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: cuda (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: han (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: adapt (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: sync (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: tuned (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: monitoring (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: basic (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA coll: sm (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA fbtl: posix (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA fbtl: ime (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA fcoll: two_phase (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA fcoll: dynamic (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA fcoll: dynamic_gen2 (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA fcoll: individual (MCA v2.1.0, API v2.0.0, Component v4.1.1)
               MCA fcoll: vulcan (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA fs: ime (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA fs: gpfs (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA fs: ufs (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA io: ompio (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA io: romio321 (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                  MCA op: avx (MCA v2.1.0, API v1.0.0, Component v4.1.1)
                 MCA osc: monitoring (MCA v2.1.0, API v3.0.0, Component v4.1.1)
                 MCA osc: pt2pt (MCA v2.1.0, API v3.0.0, Component v4.1.1)
                 MCA osc: rdma (MCA v2.1.0, API v3.0.0, Component v4.1.1)
                 MCA osc: ucx (MCA v2.1.0, API v3.0.0, Component v4.1.1)
                 MCA osc: sm (MCA v2.1.0, API v3.0.0, Component v4.1.1)
                 MCA pml: v (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA pml: monitoring (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA pml: ucx (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA pml: ob1 (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA pml: cm (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                 MCA rte: pmix (MCA v2.1.0, API v2.0.0, Component v4.1.1)
            MCA sharedfp: sm (MCA v2.1.0, API v2.0.0, Component v4.1.1)
            MCA sharedfp: lockedfile (MCA v2.1.0, API v2.0.0, Component v4.1.1)
            MCA sharedfp: individual (MCA v2.1.0, API v2.0.0, Component v4.1.1)
                MCA topo: treematch (MCA v2.1.0, API v2.2.0, Component v4.1.1)
                MCA topo: basic (MCA v2.1.0, API v2.2.0, Component v4.1.1)
           MCA vprotocol: pessimist (MCA v2.1.0, API v2.0.0, Component v4.1.1)
```

-----------------------------

## Details of the problem

Here is a complete short program that reproduces the problem:
```
#include <iostream>

#include <mpi.h>

int main(void)
{
  int requested = MPI_THREAD_MULTIPLE, provided;

  MPI_Init_thread(nullptr, nullptr, requested, &provided);

  if (provided != requested)
  {
    std::cerr << "Failed to initialize MPI with full thread support!"
              << std::endl;
    exit(1);
  }

  const size_t dim = 1024;
  void *mem = nullptr;

  MPI_Info info;

  MPI_Info_create(&info);

  MPI_Info_set(info, "same_disp_unit", "true");
  MPI_Info_set(info, "no_locks", "true");
  MPI_Info_set(info, "accumulate_ops", "same_ops");

  MPI_Win win;

  MPI_Win_allocate(
    dim * sizeof(double),
    sizeof(double),
    info,
    MPI_COMM_WORLD,
    &mem,
    &win
  );

  MPI_Win_free(&win);
  MPI_Finalize();
}
```
I get output like the following:
```
*** An error occurred in MPI_Win_allocate
*** reported by process [2289576957,0]
*** on communicator MPI_COMM_WORLD
*** MPI_ERR_WIN: invalid window
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
```
With `MPI_INFO_NULL` passed instead of `info`, there is again no issue.

**Comment by hppritcha** [comment]
able to reproduce with OMPI 4.1.1

**Comment by sekoenig** [comment]
I found that as a workaround it is possible to use `MPI_Win_set_info` after allocating the window, which does not lead to a crash and supposedly gives the desired configuration, but it would still be good to get the problem fixed in `MPI_Win_allocate`.

**Comment by hppritcha** [comment]
okay took a while to figure out what's going on here.  there's an error in the test.  the accumulate ops value should be ```same_op``` NOT ```same_ops``` according to the MPI 4 standard.  Changing the string to ```same_op``` allows the test to pass.

**Comment by sekoenig** [comment]
Argh, my bad, sorry about that!

I ran a quick test with that fix applied to my sample code and it seems to be resolved there for me also.

I thought in my actual project I observed the crash as soon as _any_ info key is set, i.e., without specifying the `accumulate_ops` at all, but I am not sure now.  Unfortunately today the cluster went offline for a two-day maintenance, so I cannot check that right now.  I would ask to still keep this open until the end of this week so I can either verify that or confirm that it was merely my mistake overall.
